File: src/server/TypeLibrary.Models/Data/Simple.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace TypeLibrary.Models.Data
{
    public class Simple
    {
        public string Id { get; set; }
        public string Name { get; set; }
        
        public string Kind => nameof(Simple);
        public string SemanticReference { get; set; }
        public virtual ICollection<Attribute> Attributes { get; set; }
        public virtual string NodeId { get; set; }

        [JsonIgnore]
        public Node Node { get; set; }
    }
}

File: src/client/typelibrary/src/models/classes/Simple.ts
import { Attribute } from "..";

export const SIMPLE_KIND = "Simple";

class Simple {
  id: string;
  name: string;
  semanticReference: string;
  attributes: Attribute[];
  nodeId: string;

  kind: string = SIMPLE_KIND;

  constructor(simple: Simple) {
    Object.assign(this, simple);
  }
}

export default Simple;
